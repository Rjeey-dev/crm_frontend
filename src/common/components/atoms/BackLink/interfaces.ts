import {History} from "history";
import {ReactNode, ReactNodeArray} from "react";
import {RouteComponentProps} from "react-router";

export interface IProps extends RouteComponentProps {
    classes: string,
    children: ReactNode | ReactNodeArray,
    history: History
}