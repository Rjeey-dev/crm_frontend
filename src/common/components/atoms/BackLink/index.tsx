import React from 'react';
import { withRouter } from 'react-router-dom';

import Link from 'common/components/atoms/Link/index';
import {IProps} from "./interfaces";

function BackLink(props: IProps) {
    return <Link href='#back' classes={props.classes} onClick={props.history.goBack}>{props.children}</Link>;
}

export default withRouter(BackLink);