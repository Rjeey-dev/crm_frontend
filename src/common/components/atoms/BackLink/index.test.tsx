import React from 'react';
// @ts-ignore
import MockRouter from 'react-mock-router';
import renderer from 'react-test-renderer';

import BackLink from './index';

it('Renders <BackLink/> with correct properties', () => {
    const output = renderer.create(<MockRouter><BackLink text='Some text' classes='class'/></MockRouter>).toJSON();

    expect(output).toMatchSnapshot();
});
