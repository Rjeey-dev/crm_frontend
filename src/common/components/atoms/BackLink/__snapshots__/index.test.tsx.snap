// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <BackLink/> with correct properties 1`] = `
<a
  className="class"
  href="#back"
>
  Some text
</a>
`;
